### PR TITLE
Make content scope control searchable

### PR DESCRIPTION
**Pull request by VP-DS**
`Internal Link:` https://app-eu.wrike.com/open.htm?id=1081727118

<details>
    <summary>Screenshots/Videos:</summary>
Images of the clear button:
Screenshot 2023-04-19 at 10 08 03
Screenshot 2023-04-19 at 10 08 13

Video showing the search functionality without the clear button:

https://user-images.githubusercontent.com/118435139/232798849-5979b00b-a019-461e-b7d1-9bf3bae55642.mov

</details>

### CODE REVIEW

File: packages/admin/cms-admin/src/contentScope/Select.tsx [hunk 1]
```diff
@@ -13,24 +15,61 @@ export interface ContentScopeSelectProps {
...
+    const filteredValues = searchable ? values.filter((item) => item.value.toLowerCase().includes(searchValue.toLowerCase())) : values;
+
     return (
-        <AppHeaderDropdown buttonChildren={value ? value.label || value.value.toUpperCase() : defaultLabel} startIcon={Icon ? <Icon /> : undefined}>
+        <AppHeaderDropdown
+            onClose={() => {
+                setTimeout(() => {
+                    setSearchValue("");
+                }, 250);
+            }}
```
[the remark this answers is not in the document]
Comment (johnnyomair) [reply packages/admin/cms-admin/src/contentScope/Select.tsx:45]: Wouldn't this need to make the component controlled? Do you prefer that?

[the remark this answers is not in the document]
Comment (VP-DS) [reply packages/admin/cms-admin/src/contentScope/Select.tsx]: Moved the clearing of the searchValue to be done in the text input. In github or gitlab the element also doesn't autoclears the value. (Also  if you use the current onOpenChange it overwrites the default logic)
Screenshot 2023-04-19 at 09 06 52

File: packages/admin/cms-admin/src/contentScope/Select.tsx [hunk 2]
```diff
@@ -13,24 +15,56 @@ export interface ContentScopeSelectProps {
...
             {(hideDropdown) => (
                 <List>
-                    {values.map(({ value: v, label: l }) => (
-                        <ListItem
+                    {searchable && (
+                        <InputBase
+                            sx={{ m: 1 }}
+                            startAdornment={<Search />}
+                            placeholder={intl.formatMessage({
+                                id: "search.input",
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/contentScope/Select.tsx:48]:
```suggestion
                                id: "contentScope.select.searchInput.placeholder",
```

The ID of the message should be more specific

**Comment by johnnyomair** [comment]
Please add a changelog entry using `npx changeset`

File: .changeset/shaggy-cherries-design.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-admin": minor
+---
+
+This change adds an optional content scope filter. This is useful if you have a very long list of different content scopes, that you cannot separate. To use it add 'searchable: true' to your ContentScopeProvider. (See how it is done in demo/admin/src/common/ContentScopeProvider.tsx)
```
Comment (johnnyomair) [inline .changeset/shaggy-cherries-design.md:5]:
```suggestion
Add support to search values in content scope controls. Add `searchable: true` to a control's config to make it searchable. See [ContentScopeProvider](demo/admin/src/common/ContentScopeProvider.tsx) for an example
```

File: packages/admin/cms-admin/src/contentScope/Select.tsx [hunk 4]
```diff
@@ -13,24 +15,56 @@ export interface ContentScopeSelectProps {
...
     return (
         <AppHeaderDropdown buttonChildren={value ? value.label || value.value.toUpperCase() : defaultLabel} startIcon={Icon ? <Icon /> : undefined}>
             {(hideDropdown) => (
                 <List>
-                    {values.map(({ value: v, label: l }) => (
-                        <ListItem
+                    {searchable && (
+                        <InputBase
+                            sx={{ m: 1 }}
+                            startAdornment={<Search />}
```
Comment (jamesricky) [inline packages/admin/cms-admin/src/contentScope/Select.tsx:46]: Please wrap icons used as input-adornments with `<InputAdornment />` like in the MUI examples: https://mui.com/material-ui/react-text-field/#icons

```suggestion
                            startAdornment={<InputAdornment position="start"><Search /></InputAdornment>}
```

**Review by johnnyomair — APPROVED** [review]

**Review by jamesricky — APPROVED** [review]

**Pull request merged**